cloudify_terminal/tasks.py

```diff
@@ -210,3 +210,82 @@ def run(*args, **kwargs):
         time.sleep(1)
 
     connection.close()
+
+
+@operation_cleanup
+def run(*args, **kwargs):
+    """main entry point for all calls"""
+    # get current context
+    ctx = kwargs.get('ctx', CloudifyContext)
+
+    if ctx.type == context.NODE_INSTANCE:
+        # Node instance
+        properties = ctx.node.properties
+        runtime_properties = ctx.instance.runtime_properties
+
+        # get ip from parent
+        try:
+            host_ip = ctx.instance.host_ip
+        except cfy_exc.NonRecoverableError:
+            host_ip = None
+
+        log_stamp = "{execution_id}_{instance_id}_{workflow_id}".format(
+            execution_id=ctx.execution_id,
+            instance_id=ctx.instance.id,
+            workflow_id=ctx.workflow_id
+        )
+
+    elif ctx.type == context.RELATIONSHIP_INSTANCE:
+        # Realationships context
+        properties = ctx.target.node.properties
+        runtime_properties = ctx.target.instance.runtime_properties
+
+        # get ip from parent
+        try:
+            host_ip = ctx.target.instance.host_ip
+        except cfy_exc.NonRecoverableError:
+            host_ip = None
+
+        log_stamp = "{execution_id}_{instance_id}_{workflow_id}".format(
+            execution_id=ctx.execution_id,
+            instance_id=ctx.target.instance.id,
+            workflow_id=ctx.workflow_id
+        )
+
+    _execute(
+        ctx=ctx,
+        properties=properties,
+        runtime_properties=runtime_properties,
+        get_resource=ctx.get_resource,
+        host_ip=host_ip,
+        log_stamp=log_stamp,
+        kwargs=kwargs
+    )
+
+
+# callback name from hooks config
+@workflow
+def run_as_workflow(*args, **kwargs):
+    # get current context
+    ctx = kwargs.get('ctx', CloudifyContext)
+
+    # check inputs
+    if len(args):
+        inputs = args[0]
+    else:
+        inputs = kwargs.get('inputs', {})
```

Comment:
Maybe we should just have `inputs` as a named kwarg?

Reply:
We have call inconsistency:
* in case run as workflow first (named) param will be cloudify context, 
* in case run as hook first param will be inputs, for preventing name collision in hooks parameters, like case when user like to use `inputs` as separate field name.